crates/libs/windows/src/core/manually_drop.rs

```diff
@@ -0,0 +1,46 @@
+/// Similar to the Rust Standard Library's `ManuallyDrop` wrapper, the `windows` crate's `ManuallyDrop`
+/// makes it easier to work with Win32 structs that lack ownership semantics but nevertheless allow
+/// callers to reference such non-blittable fields. Rust's `ManuallyDrop` is not used primarily because
+// it makes it very difficult to reference the wrapped value without taking ownership of it.
+#[repr(transparent)]
+#[derive(Debug)]
+pub struct ManuallyDrop<T: super::Abi> {
+    abi: T::Abi,
+}
+
+impl<T: super::Abi> ManuallyDrop<T> {
+    pub fn new(value: &T) -> Self {
+        Self { abi: value.abi() }
+    }
+
+    pub fn abi(&self) -> T::Abi {
+        unsafe { std::mem::transmute_copy(&self.abi) }
```

Comment:
Is `Abi` not constrained to be `Copy`? I guess we sometimes set `Abi` to `std::mem::ManuallyDrop` which would not be `Copy` even though semantically it usually should be.

Reply:
Sadly, it can't be `Copy`. That would make it a lot simple for callers. Rust requires `Copy` for fields of unions but won't let me make a `Copy` type without generating considerably more code by essentially unrolling `ManuallyDrop` types and duplicating their memory layout.